spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala

```diff
@@ -260,11 +291,24 @@ object DistriOptimizer {
         models.mapPartitions(modelIter => {
           val modelCache = modelIter.next()
           parameters.aggregrateGradientPartition()
-          parameters.gradientPartition.div(ev.fromType(finishedModelNum))
+          if (!optimMethod.isInstanceOf[EAMSGD[T]]) {
```

Comment:
For EASGD `parameters.gradientPartition` actually holds `elasticDifferencepartition` which do not need to do the average operation.